src/bin/pgcopydb/dump_restore.c

```diff
@@ -622,169 +786,48 @@ copydb_write_restore_list(CopyDataSpec *specs, PostgresDumpSection section)
 	 *   2. edit post.list to comment out lines and save as filtered.list
 	 *   3. pg_restore --section post-data --use-list filtered.list schema.dump
 	 */
-	ArchiveContentArray contents = { 0 };
-
 	if (!pg_restore_list(&(specs->pgPaths),
 						 dumpFilename,
-						 listOutFilename,
-						 &contents))
+						 listOutFilename))
 	{
 		/* errors have already been logged */
 		return false;
 	}
 
 	/* edit our post.list file now */
-	PQExpBuffer listContents = createPQExpBuffer();
-
-	if (listContents == NULL)
+	PQExpBuffer line = createPQExpBuffer();
+	if (line == NULL)
 	{
 		log_error(ALLOCATION_FAILED_ERROR);
-		destroyPQExpBuffer(listContents);
+		destroyPQExpBuffer(line);
 		return false;
 	}
 
-	/* for each object in the list, comment when we already processed it */
-	for (int i = 0; i < contents.count; i++)
+	FILE *listFile = fopen_with_umask(listFilename, "wb", FOPEN_FLAGS_W, 0644);
+	if (!listFilename)
```

Comment:
This needs to be written `if (listFile == NULL)`, and as an immediate benefit would fix the bug where you compare the `const char *listFilename` instead of the `FILE *listFile` after opening.